golem-worker-service-base/src/api_definition/http/http_api_definition.rs

```diff
@@ -17,14 +17,76 @@ use crate::parser::{GolemParser, ParseError};
 use crate::worker_binding::CompiledGolemWorkerBinding;
 use crate::worker_binding::GolemWorkerBinding;
 
-#[derive(Debug, Clone, PartialEq, Serialize, Deserialize, Encode, Decode)]
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct HttpApiDefinitionRequest {
+    pub id: ApiDefinitionId,
+    pub version: ApiVersion,
+    pub routes: Vec<Route>,
+    #[serde(default)]
+    pub draft: bool,
+}
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
 #[serde(rename_all = "camelCase")]
 pub struct HttpApiDefinition {
     pub id: ApiDefinitionId,
     pub version: ApiVersion,
     pub routes: Vec<Route>,
     #[serde(default)]
     pub draft: bool,
+    pub created_at: chrono::DateTime<chrono::Utc>,
+}
+
+impl HttpApiDefinition {
+    pub fn new(
+        request: HttpApiDefinitionRequest,
+        created_at: chrono::DateTime<chrono::Utc>,
+    ) -> Self {
+        HttpApiDefinition {
+            id: request.id,
+            version: request.version,
+            routes: request.routes,
+            draft: request.draft,
+            created_at,
+        }
+    }
+}
+
+impl From<HttpApiDefinition> for HttpApiDefinitionRequest {
+    fn from(value: HttpApiDefinition) -> Self {
+        HttpApiDefinitionRequest {
+            id: value.id,
+            version: value.version,
+            routes: value.routes,
+            draft: value.draft,
+        }
+    }
+}
+
+impl From<CompiledHttpApiDefinitionRequest> for HttpApiDefinitionRequest {
+    fn from(compiled_http_api_definition: CompiledHttpApiDefinitionRequest) -> Self {
+        HttpApiDefinitionRequest {
+            id: compiled_http_api_definition.id,
+            version: compiled_http_api_definition.version,
+            routes: compiled_http_api_definition
+                .routes
+                .into_iter()
+                .map(Route::from)
+                .collect(),
+            draft: compiled_http_api_definition.draft,
+        }
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct CompiledHttpApiDefinitionRequest {
+    pub id: ApiDefinitionId,
+    pub version: ApiVersion,
+    pub routes: Vec<CompiledRoute>,
+    #[serde(default)]
+    pub draft: bool,
```

Comment:
Compiled version doesn't need to be a request anymore. Compiled is created from a fully formed http definition

Reply:
thanks, removed